lib/components/settings/theme_selection/bloc/theme_event.dart

```diff
@@ -155,16 +155,27 @@ class DeleteCustomTheme extends ThemeEvent with HarpyLogger {
       final lightThemeId = app<ThemePreferences>().lightThemeId;
       final darkThemeId = app<ThemePreferences>().darkThemeId;
 
-      // reset selection when deleting a selected theme
-      if (themeId == lightThemeId || themeId == darkThemeId) {
+      if (themeId >= lightThemeId || themeId >= darkThemeId) {
         bloc.add(ChangeTheme(
-          lightThemeId: themeId == lightThemeId ? 0 : null,
-          darkThemeId: themeId == darkThemeId ? 0 : null,
+          lightThemeId: _adjustedThemeId(lightThemeId),
+          darkThemeId: _adjustedThemeId(darkThemeId),
           saveSelection: true,
         ));
       }
     }
   }
+
+  // reset selection when deleting a selected theme
+  // or decrement if deleted id preceded selected id
```

Comment:
Let's give this method proper dart doc, for example:

```suggestion
  /// Returns the new `themeId` of the currently selected theme.
```